src/coreclr/jit/fgprofile.cpp

```diff
@@ -2336,9 +2336,35 @@ PhaseStatus Compiler::fgPrepareToInstrumentMethod()
     // in Tier1 so the profile will never be consumed. Thus, let's avoid unnecessary probes.
     if (minimalProfiling && (info.compFlags & CORINFO_FLG_INTRINSIC) != 0)
     {
-        fgCountInstrumentor     = new (this, CMK_Pgo) NonInstrumentor(this);
-        fgHistogramInstrumentor = new (this, CMK_Pgo) NonInstrumentor(this);
-        return PhaseStatus::MODIFIED_NOTHING;
+        bool shouldBeInstrumented = false;
+
+        // Some intrinsics should still be instrumented.
```

Comment:
Maybe say a bit more about why these particular intrinsics are the ones that should still be instrumented? In particular, if somebody adds a new intrinsic, how will they know if it should go on this list or not?

Reply:
Can we filter it primarily based on whether or not its recursive?

In general things that are `recursive` are considered "must expand" and never have a meaningful implementation, while things that are non-recursive are only "might expand" and it can depend on factors such as optimization level, what the inputs are, what the hardware supports, etc.

We could then still have this `switch` to handle the other cases and we could default to instrumenting with some selective logic for things we still don't want to instrument (say `Math.Floor` when Sse4.1 is supported).

-- This would also be a motivation to finish up the `S.R.CS.Unsafe` work for example, as we were planning on making those recursive once that was completed.

Reply:
@AndyAyersMS added comments. We might want to compose a single .h table where we'll list all these properties for each intrinsics (e.g. display name, instrumentability, etc)

Reply:
> Can we filter it primarily based on whether or not its recursive?

Not really unless you go and implement them all in Mono too (and interp) 🙂 
The existing Tier0 must-expand won't hit this path because Tier0 will inline them so they won't be instrumented anyway